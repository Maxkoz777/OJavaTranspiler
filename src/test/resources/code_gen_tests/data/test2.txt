class ClassName extends AnyValue is
  var Min : Integer
  var Max : Integer

  method MaxInt(a: Integer) : Integer is
   var max : Integer.Min
   var i : Integer.Plus(1)
   while i.Less(a.Length) loop
   if a.get(i).Greater(max) then max := a.Plus(i) end
   i := i.Plus(1)
   end
   return max
  end
end
