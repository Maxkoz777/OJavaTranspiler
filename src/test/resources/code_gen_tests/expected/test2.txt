package com.example.transpiler.generated;

public class ClassName extends AnyValue {

    Integer Min;

    Integer Max;

    public Integer MaxInt(Integer a) {
        var max = Integer.Min;
        var i = Integer.Plus(1);
        while (i.Less(a)) {
            if (a.get(i).Greater(max)) {
                max = a.Plus(i);
            }
            i = i.Plus(1);
        }
        return max;
    }
}
